Src/bat_charge.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * bat_charge.c
+ *
+ *  Created on: Sep, 26 2020
+ *      Author: Luke Oxley
+ */
+#include "bat_charge.h"
+
+uint8_t temp_array[READ_MSG_SIZE];
+uint8_t read_byte = 0;
+uint8_t write_data[WRITE_MSG_SIZE + 1];
```

Comment:
make sure to prefix global variables with a `g_` that way it's clear that they are global variables and not undefined stack variables